### ISSUE TITLE
Tune Bicycle Penalties / Preferences

### ISSUE THREAD

**Issue by MoKob** [Project-OSRM/osrm-backend#4237]
We have a lot of mechanics in place by now that allow fine tuning the bicycle profile to prefer certain types of roads. The big difficulty I see here is that bike riding seems to depend a lot more on personal preferences than going by car.

Given the amount of feedback we have received over the year, it could be a good time to revisit how we define our preferences in the bike profiles and perform some fine tuning.

I see many knobs we could tune:

 - surface speeds
 - road types
 - dedicated bike lanes
 - trails
 - scenic value
 - traffic lights / crossings
 - complexity of intersections
 - ...

**Comment by emiltin** [comment]
might be worth considering variations  of the bike profile. for example you mention scenic value - it might be more attractive when you're going home from work and have a bit extra time, than in the morning when you need to catch a meeting. it's hard to cover all these use cases in a single profile.

**Comment by emiltin** [comment]
many of the current settings are from the time when you could only change the speed, since there was not weight to adjust. we should consider for each way characteristic how much it should affect speed vs. weight

**Comment by danpat** [comment]
I agree with @emiltin, we should probably consider multiple bike profiles - there are vastly different needs from different cyclists.

Here's a list to start:

  - safety profile - focused on low-speed riders, children, and non-sweaty recreation - prioritize bike paths over roads, bias towards bike lanes even if it takes you a bit further
  - commuter profile - for folks that ride daily - prefer bike lanes, and bias towards a faster route
  - roadie profile - for folks wanting to do a weekend ride, who ride fast

Because hills are such a big deal, I think we should also bundle some scripts to download DEM data (SRTM or something?) that the profiles are actually built to use - I think this would go a long way to improving the bike profile just by itself.

**Comment by emiltin** [comment]
we also work with 'greenest' bike routes, which prefer parks, green areas, paths along lakes, etc. might. it's currently done with postgis queries from the lua profile, but it could maybe be done easier with raster queries, if a suitable dataset is available.

**Comment by emiltin** [comment]
cargo bikes is another one profile we use at ibikecph.dk. it avoids all steps and has a higher penalty for bad surfaces, can't bass bicycle barriers,  has a different speed, etc.
